pkg/stacks/templates/engines/helm2.go

```diff
@@ -0,0 +1,266 @@
+
+	"github.com/crossplaneio/crossplane/apis/stacks/v1alpha1"
+)
+
+// Helm2EngineRunner creates resources from files which are specified in the helm2 format
+type Helm2EngineRunner struct {
+	Log logr.Logger
+}
+
+const (
+	spec             = "spec"
+	valuesYaml       = "values.yaml"
+	engineImageName  = "crossplane/resource-engine-helm2:latest"
+	kubectlImageName = "crossplane/resource-engine-kubectl:latest"
+
+	engineCfgVolumeName = "engine-configuration"
+	engineCfgDir        = "/usr/share/engine-configuration/"
+
+	stackVolumeName = "stack-configuration"
+	stackDestDir    = "/usr/share/input/"
+
+	resourceCfgVolumeName = "resource-configuration"
+	resourceCfgDestDir    = "/usr/share/resource-configuration/"
+)
+
+var (
+	// TODO this file name should not be hard-coded
+	engineCfgFile = filepath.Join(engineCfgDir, valuesYaml)
+)
+
+// CreateConfig creates a config map to be consumed by an execution of the helm2 engine.
+// When a behavior executes, the resource engine is configured by the
+// object which triggered the behavior. This method encapsulates the logic to
+// create the resource engine configuration from the object's fields.
+// TODO it seems as though a lot of the transformation logic is probably reusable
+func (her *Helm2EngineRunner) CreateConfig(claim *unstructured.Unstructured, hc *v1alpha1.HookConfiguration) (*corev1.ConfigMap, error) {
+	// yamlyamlyamlyamlyaml
+	// TODO if spec is missing, this won't work very well
+	s, ok := claim.Object[spec]
+
+	if !ok {
+		her.Log.V(0).Info("Spec not found on claim; not creating engine configuration", "claim", claim)
+	}
+
+	her.Log.V(0).Info("Converting configuration", "spec", s)
+	configContents, err := yaml.Marshal(s)
+
+	her.Log.V(0).Info("Configuration contents as yaml", "configContents", configContents)
+
+	if err != nil {
+		her.Log.Error(err, "Error marshaling claim spec as yaml!", "claim", claim)
+		return nil, err
+	}
+
+	// Underneath, the yamler uses https://godoc.org/encoding/json#Marshal,
+	// which means that the bytes are UTF-8 encoded
+	// Theoretically we could get better performance by using a binary config
+	// map, but having a string makes it better for humans who may want to observe
+	// or troubleshoot behavior.
+	stringConfigContents := string(configContents)
+
+	// TODO get the engine type from the configuration
+	engineType := hc.Engine.Type
+
+	// TODO engine type should have a bit more structure;
+	// probably better to use an enum type pattern, with an
+	// engine name and its corresponding configuration file
+	// name in the same object
+	configKeyName := ""
+
+	if engineType == Helm2EngineType {
+		configKeyName = valuesYaml
+	}
+
+	configName := string(claim.GetUID())
+	generatedMap, err := generateConfigMap(configName, configKeyName, stringConfigContents, her.Log)
+
+	if err != nil {
+		her.Log.V(0).Info("Error generating config map!", "claim", claim, "error", err)
+		return nil, err
+	}
+
+	generatedMap.SetNamespace(claim.GetNamespace())
```

Comment:
That could be a potential change later, but currently, the engine implementation knows how to create the config map and run the underlying resource engine, and the caller doesn't know how it chooses to do that, so I feel it makes more sense to keep it inside the engine for now. Otherwise, we'd have to ask the question of whether the engine caller should set the namespace for any jobs which are run. Which might make sense, but then it has to know about jobs, and it'd be a bigger change.